### PR TITLE
Use beautifulsoup4 instead of beautifulscraper

### PR DESCRIPTION
**Pull request by Rafiot**
bs4 supports python2.7 and 3.X

### CODE REVIEW

File: tools/scraper.py [hunk 1]
```diff
@@ -2,35 +2,57 @@
...
-import beautifulscraper
+import urllib.parse
 
+from bs4 import BeautifulSoup
+import requests
 
 ROOT_ZONE_DB = 'http://www.iana.org/domains/root/db'
 SLEEP = 0
 
+IP_ASSIGNATIONS = 'https://www.iana.org/assignments/ipv4-address-space/ipv4-address-space.csv'
```
Comment (kgaughan) [inline tools/scraper.py]: Could you separate the IP assignment stuff out for now?

File: tools/scraper.py [hunk 2]
```diff
@@ -2,35 +2,57 @@
...
+    for prefix in no_server:
+        print('; No record for %s' % prefix)
+
 
 def main():
     """
-    Scrape IANA's root zone database.
-
-    The scraped data is written to standard output.
+    Scrape IANA's root zone database and write the resulting configuration
```
Comment (kgaughan) [inline tools/scraper.py]: Could you revert to the old docstring?

File: tools/scraper.py [hunk 3]
```diff
@@ -2,35 +2,57 @@
 A scraper which pulls zone WHOIS servers from IANA's root zone database.
 """
 
+import csv
 import logging
 import socket
+import io
```
Comment (kgaughan) [inline tools/scraper.py]: Could you move this before the 'logging' import?

File: tools/scraper.py [hunk 4]
```diff
@@ -43,9 +65,10 @@ def main():
 
         time.sleep(SLEEP)
 
-        zone_url = urlparse.urljoin(ROOT_ZONE_DB, link.attrs['href'])
+        zone_url = urllib.parse.urljoin(ROOT_ZONE_DB, link.attrs['href'])
```
Comment (kgaughan) [inline tools/scraper.py]: `urllib.parse` is Python 3 only.

Pulling in either `six` or `future` might be worthwhile to bridge the gap. Alternatively, you could add a module called `uwhoisd.compat` with the following:

    """
    Python 2/3 compatibility layer.
    """

    try:
        from urllib.parse import urljoin
    except ImportError:
        from urlparse import urljoin

And then pull `urlparse` in from that.

File: requirements-test.txt [hunk 5]
```diff
@@ -1,5 +1,8 @@
 -e .
 
+# scrap IANA
```
Comment (kgaughan) [inline requirements-test.txt]: Typo: missing an 'e' in 'scrape'.

File: requirements-test.txt [hunk 6]
```diff
@@ -1,5 +1,8 @@
...
+bs4
```
Comment (kgaughan) [inline requirements-test.txt]: I'd actually move this to `requirements-dev.txt`.

Might as well add a dependency on *requests* after it too.

File: tools/scraper.py (hunk 1 above)
Comment (Rafiot) [inline tools/scraper.py]: Done

File: tools/scraper.py (hunk 2 above)
Comment (Rafiot) [inline tools/scraper.py]: done

File: tools/scraper.py (hunk 3 above)
Comment (Rafiot) [inline tools/scraper.py]: This one was only used by the IP assignations scrapping

File: tools/scraper.py (hunk 4 above)
Comment (Rafiot) [inline tools/scraper.py]: As it is an external script, I think it is better to add the import in the script directly

File: requirements-test.txt (hunk 6 above)
Comment (Rafiot) [inline requirements-test.txt]: done

**Pull request merged**